sdks/python/apache_beam/io/aws/clients/s3/fake_client.py

```diff
@@ -0,0 +1,235 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import datetime
+import time
+
+from apache_beam.io.aws.clients.s3 import messages
+
+
+class FakeFile(object):
+
+  def __init__(self, bucket, key, contents, etag=None):
+    self.bucket = bucket
+    self.key = key
+    self.contents = contents
+
+    self.last_modified = time.time()
+
+    if not etag:
+      self.etag = '"%s-1"' % ('x' * 32)
+    else:
+      self.etag = etag
+
+  def get_metadata(self):
+    last_modified_datetime = None
+    if self.last_modified:
+      last_modified_datetime = datetime.datetime.utcfromtimestamp(
+          self.last_modified)
+
+    return messages.Item(self.etag,
+                         self.key,
+                         last_modified_datetime,
+                         len(self.contents))
```

Comment:
It seems that the problem is a missing argument here (`mime_type`). : ) @MattMorgis @tamera-lanham

Reply:
Hey! Sorry it's been a bit since I've been in this codebase. A couple notes:
- In the constructor for `messages.Item`, `mime_type` is optional and defaults to `None` (skds/python/apache_beam/io/aws/clients/s3/messages.py:121), so it's surprising that this error is appearing. When I run the tests locally the behavior is what I'd expect - the `mime_type` gets set to `None` and the test passes, so I don't know why it isn't doing that in CI. I can't replicate the failure in my environment (which is Dockerized, if you want to try building it yourself!)
- The reason for using the default of `None` is because it was kind of onerous to build mime type support into the fake client. The only time mime type behavior is tested is in s3io_test.py, in `test_file_mime_type`, and that passes against the real client and is skipped for the fake. I like that metadata from the fake client comes out with a mime type of `None`, to indicate to whoever consumes the fake client that mime types aren't supported. Would you be ok with a default of `None` here instead of `application/octet-stream`? Whatever the default is, it shouldn't show up in any test anyway (except the one that we skip for the mock anyway, in which case we're skipping it)

Reply:
That's totally fine by me : ) - you're right, it seems that the default value for mime type in  `Item` was not set to None. I am okay with letting it be None. Thanks for taking a look.